### PR TITLE
DLQ config parsing

### LINKED ISSUE #716: DLQ: Pipeline config file
Update pipeline config parser to parse and provision DLQ. Bump config version because of the change.

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

This PR changes the config parser to also parse DLQ configs.

Fixes #716

### Quick checks:

- [X] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [X] There is no other [pull request](https://github.com/ConduitIO/conduit/pulls) for the same update/change.
- [X] I have written unit tests.
- [X] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: pkg/pipeline/service.go [hunk 1]
```diff
@@ -193,6 +193,36 @@ func (s *Service) Update(ctx context.Context, pipelineID string, cfg Config) (*I
...
+	pl, err := s.Get(ctx, pipelineID)
+	if err != nil {
+		return nil, err
+	}
+
+	if cfg.Plugin == "" {
+		return nil, cerrors.New("DLQ plugin must be provided")
+	}
+	if cfg.WindowSize < 0 {
+		return nil, cerrors.New("DLQ window size must be non-negative")
```
Comment (maha-hajja) [inline pkg/pipeline/service.go:207]: what does it mean if window size is 0?

File: pkg/provisioning/parse.go [hunk 2]
```diff
@@ -44,6 +44,14 @@ type PipelineConfig struct {
 	Description string                     `yaml:"description"`
 	Connectors  map[string]ConnectorConfig `yaml:"connectors,omitempty"`
 	Processors  map[string]ProcessorConfig `yaml:"processors,omitempty"`
+	DLQ         DLQConfig                  `yaml:"dead-letter-queue"`
+}
+
+type DLQConfig struct {
+	Plugin              string            `yaml:"plugin"`
+	Settings            map[string]string `yaml:"settings"`
+	WindowSize          *int              `yaml:"window-size"`
```
Comment (maha-hajja) [inline pkg/provisioning/parse.go:53]: is there a specific reason to have it as a pointer?

File: pkg/pipeline/service.go (hunk 1 above)
Comment (lovromazgon) [reply pkg/pipeline/service.go:207]: Window size 0 means that no records are monitored, all nacked records are rerouted to the DLQ without a limit so the pipeline doesn't stop.

File: pkg/provisioning/parse.go (hunk 2 above)
Comment (lovromazgon) [reply pkg/provisioning/parse.go:53]: Yes, setting the window size to 0 is a valid thing to do, while if it's not set the default window size is 1, so we have to be able to differentiate if the window size was explicitly set to 0 or if it is empty.

Comment (maha-hajja) [reply pkg/provisioning/parse.go:53]: oh makes sense! thanks for the explanation

**Pull request merged**